src/java-runtime/java/mono/android/MonoPackageManager.java

```diff
@@ -35,6 +35,11 @@ public static void LoadApplication (Context context, ApplicationInfo runtimePack
 				);
 				context.registerReceiver (new mono.android.app.NotifyTimeZoneChanges (), timezoneChangedFilter);
 
+				android.content.IntentFilter dumpTimingDataFilter = new android.content.IntentFilter (
+						"mono.android.app.DUMP_TIMING_DATA"
+				);
+				context.registerReceiver (new mono.android.app.DumpTimingData (), dumpTimingDataFilter);
```

Comment:
Is there any problem to put this in all `Release` apps? I know we probably want to *be able* to get the timings in `Release` mode, but you'd be able to run this broadcast on customers' apps on Google Play.

Reply:
> Is there any problem to put this in all `Release` apps? I know we probably want to _be able_ to get the timings in `Release` mode, but you'd be able to run this broadcast on customers' apps on Google Play.

It would be completely harmless then, [see here](https://github.com/xamarin/xamarin-android/pull/6970/files#diff-c2b317c53337d77015d6dc79f6a41f5fab23908c44e83ad7ef16be79b66abeb7R2533).

Reply:
This might also be useful for customers who want to profile their app I guess.

But I do wonder if there is a way we can enable this at built time somehow, so generate the java code which adds the Intent .

Reply:
The only way would be to have a 3rd build for both the native runtime and the managed+java parts. We could probably put a runtime condition on the above code, but it would have to be either a system property check or a Bundle key lookup, neither of which is free. The native code would have to be compiled with appropriate options, OTOH, since the timing code is sprinkled all around our code base **and** inlined (which is responsible for the `libmonodroid.so` size increase). Personally, I would prefer to have a separate "timing" build (Release only, Debug can include the timing infrastructure) as it would have even less impact on the actual measurements, since we wouldn't need runtime checks whether timing is enabled or not. However, that would change the way we've been using this infrastructure since very long time ago, and it would also make the build process more (conceptually) complicated as we'd have to introduce a new "supported" configuration, say, `ReleaseTiming`. But, maybe, it's not such a bad idea? If it were implemented, the "standard" `Release` build could contain no timing code at all.

Reply:
I was just wondering if we can put something like this in the manifest and remove the need to hardcode it?

```
<receiver android:name="mono.android.app.DumpTimingData">
    <intent-filter>
        <action android:name="mono.android.app.DUMP_TIMING_DATA"/>
    </intent-filter>
</receiver>
```

If this is the case we can expose an MSbuild property and have the generateJavaStubs or manifest merger inject the required receiver element.

Reply:
OK so I tested the following.

Add the code below to a `Test.java` source file

```java
package mono.android.app;
import android.content.BroadcastReceiver;
import android.content.Intent;
import android.content.Context;
import android.util.Log;


public class Test extends BroadcastReceiver
{
    public Test ()
    {

    }
    @Override
    public void onReceive(Context context, Intent intent)
    {
        Log.d ("DEBUG", "Got a broadcast!!!! DEBUG!!!");
    }
}
```

Add the following to your csproj

```xml
<ItemGroup>
   <AndroidJavaSource Include="Test.java" />
</ItemGroup>
```

Add the following to your `application` element in your `AndroidManifest.xml`

```xml
	<receiver android:name="mono.android.app.Test">
		<intent-filter>
			<action android:name="mono.android.app.DUMP_TIMING_DATA"/>
		</intent-filter>
	</receiver>
```

Build and run your app (make sure the app is actually running).

Run 

```
 adb shell am broadcast -a mono.android.app.DUMP_TIMING_DATA <packagename>
```

and you should see the log message from the code above in `adb logcat`. 
This is with no C# code to register the reciever. So we should be able to generate this reciever at build time based on a MSBuild property that the end user can set. This will mean we don't need to always include the reciever in all builds.